### DISCUSSION TITLE
Entity Question

### DISCUSSION THREAD

**Discussion by gregeeh** [blakeblackshear/frigate#4771] [General]
Hi all,

I have Frigate in HA and I was wondering what was the difference between Motion and Detect Switch Entities. If someone could explain it would be appreciated.

TIA

https://i.imgur.com/UY3W08W.jpeg

**Comment by NickM-27** [comment]
The detect switch enables / disables object detection. 

Motion is for motion detection. Motion detection is required for object detection but it also offers other functions like motion based recording retention and the motion sensor in HA so it has its own switch.

**Reply by gregeeh** [reply]
Thanks for your reply @NickM-27.  Are you saying the Detect Switch is for detecting objects like person, dogs etc.  But Motion will detect movement like trees blowing in the wind, leafs falling onto the ground etc.

**Reply by NickM-27** [reply]
Yes, more or less 👍

**Reply by gregeeh** [reply]
Sorry, one more question.

You say "Motion detection is required for object detection ".  So to have Object Detection, like person, Motion Detection should be enabled as well.

The reason I ask is I have been just turning ON Object Detection and it has been detecting people.

Sorry, but I'm still confused.

**Reply by NickM-27** [reply]
Yes, frigate only looks for objects in places where motion is detected first. To be able to detect objects, motion detection must be enabled. (it also won't let you disable motion if detect is enabled)

**Reply by gregeeh** [reply]
Well, something must be very wrong with my setup.  If I turn ON Detect and leave Motion OFF I get objects detected.  If I turn ON Motion, with or without Detect ON, it turns OFF immediately.

**Reply by NickM-27** [reply]
Odds are you're not on the latest version of frigate so the motion switch is doing nothing

**Reply by gregeeh** [reply]
Debug Page is showing:

0.10.1-83481AF

**Reply by NickM-27** [reply]
Yes, 0.11 is the latest version. If you're using docker the tag changed from `stable-amd64` to just `stable`

**Reply by gregeeh** [reply]
Yes, in using docker and the old tag, so will update...

Have updated to version 0.11.1-2EADA21 and now the Motion Switch has defaulted to being ON.  Where before I could not get it to stay ON.  Maybe detection will work better now?

**Reply by NickM-27** [reply]
Nothing has changed with the way detection works, just an option to toggle motion. I'd seriously recommend reading the release notes if you haven't already.

**Reply by gregeeh** [reply]
Thank you, will do.

**Reply by fliespl** [reply]
Does anyone know what could be issue with motion binary sensor being always "cleared" while detections are working?

**Reply by NickM-27** [reply]
you should make your own support discussion, provide examples of what you are referring to
